### ISSUE TITLE
Mirabox Stream Dock Support 

### ISSUE THREAD

**Issue by Irvine81** [pre-martin/StreamDeckSimHubPlugin#117]
Hallo Martin, 

anbei der Link zur Software für das Stream Dock:
Key123.vip/win

Vielleicht hast du eine Möglichkeit es dort auch zum laufen zu bringen. 
Wäre eine günstige Alternative zu dem Elgato Stream Deck. 

Danke schon mal,
Erwin

**Comment by pre-martin** [comment]
https://sdk.key123.vip/en/guide/overview.html

**Comment by pre-martin** [comment]
I am working on two different topics, so unfortunately, there is no progress. But if you think that not much is missing, I can take a closer look at it again.

**Comment by MagicKoss** [comment]
hi, yes I have already imported the plugin manually and it works, but on my mirabox N4 the display/encoder part doesn't work

**Comment by MagicKoss** [comment]
Hi, please can you create/modify this plugin for this device:
Stream Dock MBox 293N4
https://sdk.key123.vip/en/guide/overview.html#stream-dock-mbox-293n3
THANKS

**Comment by Heart-State** [comment]
<img width="1873" height="1036" alt="Image" src="https://github.com/user-attachments/assets/24d029d6-e75f-4d82-b0e5-25dce15df3ce" />
I noticed that there is a folder called libs missing from the pi folder in this plugin package. After adding this folder, the plugin can run normally except for the knob.

**Comment by Heart-State** [comment]
hint:[net.planetrenner.simhub.streamDeckPlugin](https://github.com/pre-martin/StreamDeckSimHubPlugin/releases/download/v1.13.69/net.planetrenner.simhub.streamDeckPlugin)
You can change the file suffix .streamDeckPlugin to .zip and then modify the contents after decompression. After modification, re-compress it into a .zip file and then change the .zip suffix to .sdPlugin to turn it into a StreamDock plug-in installation package.

**Comment by stevenbrookes** [comment]
Just FYI this manifest file works fine with VSD Inside 293S.  I had to move the plugin directory to AppData\Roaming\HotSpot\StreamDock\plugins\net.planetrenner.simhub.sdPlugin, then restart VSD Craft.]

Only issue is the screen doesn't work.  Any thoughts on how that might be setup?

[manifest.json](https://github.com/user-attachments/files/24372939/manifest.json)

**Comment by pre-martin** [comment]
Thank you for your feedback. So many clones, so many different "dialects"...

I will take a look into supporting the Stream Dock 293S.

**Comment by stevenbrookes** [comment]
Here's the manifest from weather plugin that can draw onto the display area on the right of the 293S.  It looks like the action is Information.  The display area allow 3 vertical square display only buttons to be added.  Do you think the Generic button could be persuaded to display here.  It would make for a really useful display for brake bias, traction control, ABS settings etc.

[manifest.json](https://github.com/user-attachments/files/24401390/manifest.json)

**Comment by pre-martin** [comment]
@stevenbrookes The manifest entry for the Generic Button looks like:
```
    {
      "Icon": "images/actions/generic-button",
      "Name": "Generic Button",
      "PropertyInspectorPath": "pi/genericButton.html",
      "States": [
        {
          "Image": "images/icons/generic-button"
        }
      ],
      "Controllers": [
        "Encoder",
        "Keypad"
      ],
      "Encoder": {
        "layout": "layout/dial-fullscreen.json"
      },
      "Tooltip": "Generic Button",
      "SupportedInMultiActions": false,
      "UUID": "net.planetrenner.simhub.generic-button"
    },
```

You can try to add `"Information"` to the list of `Controllers`:
```
      "Controllers": [
        "Encoder",
        "Keypad",
        "Information"
      ],
```

**Edit:** Does not work. Maybe the button is not receiving the same events as an "Encoder" action. I have to dig deeper.....

**Comment by pre-martin** [comment]
Proof of concept for the information area of a Stream Dock 293S:

![Image](https://github.com/user-attachments/assets/07f1f32b-250d-4c88-a46b-200952e7f32c)

Unfortunately, I am currently injured after a sports accident, so I can only work on the computer for very short periods. It will take a while longer...

**Comment by stevenbrookes** [comment]
Wow looking very nice.  Best wishes recovering soon.

**Comment by Bram-Brooks** [comment]
Yes please! I have your latest plugin version mostly working with the Ajazz AKP135E with Opendeck (another incredible open source app for streamdecks and clones). The main bug is that the images stop working after reboot or restart of Opendeck. It does communicate fine with Simhub initially tho, and the states are displayed with your provided images without any issues.

Really hoping you'll consider supporting Opendeck as well, so these clones can really shine. Cause your work is incredible. Thanks for all you do Martin!

**Comment by pre-martin** [comment]
For the moment, I will attach a Zip file for the Stream Dock to every release, starting with version 2.3. The zip file contains a specific manifest for the Stream Dock to also support the "Information" area (see screencast above). The Stream Dock version is not yet included in the installer.

**Comment by stevenbrookes** [comment]
Excellent progress. Thank you for your efforts.
